src/Content/Post/Factory/PostMedia.php

```diff
@@ -48,24 +49,24 @@ public function createFromTableRow(array $row)
 	{
 		return new Entity\PostMedia(
 			$row['uri-id'],
-			$row['url'] ? new Uri($row['url']) : null,
+			UtilNetwork::createUriFromString($row['url']) ?: '',
```

Comment:
This is wrong too, the parameter expects a `UriInterface` object with no fallback. Is there a good reason to store post-media with an invalid URL? If not, it should be handled during creation, not when we retrieve a record from the database that's supposedly valid.

Reply:
When - for whatever reason - invalid data was stored there, this will create a completely empty timeline. Of course we should check when storing (and from my expectation we do), but better safe than sorry.

Reply:
There can be no default here, since we can store a lot of different media types.

Reply:
Indeed, but since we currently allow to store garbage in the database, we need a default URI to keep data integrity in the script. It doesn't matter what media type it is, we can't do anything with the invalid URI anyway.

Reply:
See my new solution. I think that this is the best way to handle it.